### PR TITLE
chore: Remove FilterBox.less

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
Removes the FilterBox.less file from the FilterBox component.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
No visual changes

### TESTING INSTRUCTIONS
1. Create a FilterBox chart
2. Add filters
3. Make sure no visual changes can be observed
4. Add the FilterBox to a Dashboard
5. Open and set all the filters
6. Make sure no visual changes can be observed

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/visualizations/FilterBox/FilterBox.jsx [hunk 1]
```diff
@@ -409,37 +439,56 @@ class FilterBox extends React.PureComponent {
     return filtersFields.map(filterConfig => {
       const { label, key } = filterConfig;
       return (
-        <div key={key} className="m-b-5 filter-container">
+        <StyledFilterContainer key={key} className="filter-container">
```
Comment (geido) [inline superset-frontend/src/visualizations/FilterBox/FilterBox.jsx:442]: This className is used by other less files. We should probably remove it once all the less files have been removed from the codebase

File: superset-frontend/src/visualizations/FilterBox/FilterBox.jsx [hunk 2]
```diff
@@ -91,6 +97,30 @@ const defaultProps = {
   instantFiltering: false,
 };
 
+const StyledFilterContainer = styled.div`
+  display: flex;
```
Comment (michael-s-molina) [inline superset-frontend/src/visualizations/FilterBox/FilterBox.jsx:101]: [No need to use theme the hard way](https://github.com/apache/superset/wiki/Emotion-Styling-Guidelines-and-Best-Practices#no-need-to-use-theme-the-hard-way) 😉

File: superset-frontend/src/visualizations/FilterBox/FilterBox.test.jsx [hunk 3]
```diff
@@ -46,9 +47,11 @@ describe('FilterBox', () => {
           },
         ]}
         origSelectedValues={{}}
+        theme={supersetTheme}
```
Comment (michael-s-molina) [inline superset-frontend/src/visualizations/FilterBox/FilterBox.test.jsx:50]: Can we use `styledShallow` instead of importing the theme?

Comment (geido) [reply superset-frontend/src/visualizations/FilterBox/FilterBox.test.jsx]: Hey @michael-s-molina thanks for the suggestion. I had to use `styledMount` instead. I updated the PR. Thank you!

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Pull request merged**